codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -814,6 +837,12 @@ public void operationComplete(ChannelFuture future) throws Exception {
                 }
             });
         }
+        // if closeListener != null this means we have already initiated graceful closure. doGracefulShutdown will apply
+        // the gracefulShutdownTimeoutMillis on each invocation, and it is possible the timeout value was changed
```

Comment:
@ejona86 - there is some complexity related to preserving existing functionality and `gracefulShutdownTimeoutMillis` management (which we should re-asses for 5.0).